src/gg.regression.unity.experimental.runespawn/Editor/Scripts/EditorCommandServer.cs

```diff
@@ -0,0 +1,286 @@
+using UnityEngine;
+using UnityEditor;
+using System;
+using System.IO;
+using System.Net;
+using System.Net.Sockets;
+using System.Text;
+using System.Threading;
+using System.Collections.Generic;
+
+[InitializeOnLoad]
+public class EditorCommandServer
+{
+    // Enumeration for command types
+    public enum CommandType
+    {
+        Start,
+        Stop,
+        GetScripts,
+        Compile,
+        GetCompilationMessages
+    }
+
+    // Configuration: Read from environment variables with defaults
+    private static readonly int Port = GetPort();
+    private static readonly string AuthToken = GetAuthToken();
+
+    private static Thread listenerThread;
+    private static TcpListener listener;
+    private static volatile bool stopThread = false;
+    private static readonly Queue<EditorCommand> commandQueue = new Queue<EditorCommand>();
+
+    static EditorCommandServer()
```

Comment:
(nit for future) Just copying a comment I had from my other PR review in the Runespawn repo - usually we follow a singleton pattern rather than static variables and methods. This file can serve as an example of this pattern: https://github.com/Regression-Games/RGUnityBots/blob/162ab3b57c0eaad5ed57a68ed205aa15bebd6ad8/src/gg.regression.unity.bots/Runtime/Scripts/RGOverlay/RGSequenceManager.cs#L38

Reply:
Noted! Singleton does makes more sense.